crates/flux-middle/src/rty/mod.rs

```diff
@@ -157,10 +157,12 @@ pub enum Sort {
     Tuple(List<Sort>),
     Func(FuncSort),
     App(SortCtor, List<Sort>),
+    Var(usize),
```

Comment:
Maybe we can make a wrapper over this `usize` (akin to [`ParamTy`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/struct.ParamTy.html)) and write some documentation around binders

```rust
enum SortVar {
    pub index: usize
}
```

or
 
```rust
newtype_index! {
    pub struct SortVar { }
}
```